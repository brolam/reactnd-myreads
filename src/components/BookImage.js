import React from 'react';
import PropTypes from 'prop-types'

function BookImage(props) {
    return (
        <div className="book-cover" style={{ width: 120, height: 190, backgroundImage: `url("https://books.google.com/books/content?id=${props.bookId}&printsec=frontcover&img=1&zoom=1")` }}></div>
    )
}
BookImage.propTypes = {
    bookId: PropTypes.string.isRequired,
}
export default BookImage