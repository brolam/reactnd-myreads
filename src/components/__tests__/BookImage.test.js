import React from 'react';
import BookImage from '../BookImage.js';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

test('Renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BookImage bookId="nggnmAEACAA" />, div)
})

test('Last Snapshot', () => {
    const bookImage = renderer.create(
        <BookImage bookId="nggnmAEACAA" />
      );
      let tree = bookImage.toJSON();
      expect(bookImage).toMatchSnapshot();
})