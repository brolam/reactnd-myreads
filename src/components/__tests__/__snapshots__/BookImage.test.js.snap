// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Last Snapshot 1`] = `
<div
  className="book-cover"
  style={
    Object {
      "backgroundImage": "url(\\"https://books.google.com/books/content?id=nggnmAEACAA&printsec=frontcover&img=1&zoom=1\\")",
      "height": 190,
      "width": 120,
    }
  }
/>
`;
